crates/verifier/src/error/stacktrace.rs

```diff
@@ -0,0 +1,218 @@
+use std::fmt;
+
+use cranelift_entity::packed_option::PackedOption;
+use sonatina_ir::{
+    function::DisplaySignature,
+    global_variable::DisplayGlobalVariable,
+    insn::DisplayInsn,
+    module::{DisplayCalleeFuncRef, FuncRef},
+    types::{CompoundType, DisplayCompoundType, DisplayType},
+    value::DisplayArgValue,
+    Block, Function, GlobalVariable, Insn, Type, Value,
+};
+
+#[derive(Debug, Clone, Copy)]
+pub struct Stacktrace {
+    func: PackedOption<FuncRef>,
+    block: PackedOption<Block>,
+    insn: PackedOption<Insn>,
+    callee: PackedOption<FuncRef>,
+    value: PackedOption<Value>,
+    gv: PackedOption<GlobalVariable>,
+    ty: PackedOption<Type>,
+    cmpd_ty: PackedOption<CompoundType>,
+}
```

Comment:
The current name `Stacktrace` might be misleading. Typically, a "stack trace" refers to call stack information, but this struct holds data like `func`, `block`, `insn`, and other IR-related entities. 

Given that the output resembles IR or execution context details:
```
0: i8
1: 0.i8
2: v2.i32 = add 28.i32 0.i8;
3: block0
4: func public %test_func() -> ()
```

It seems more like an IR trace or execution context rather than a typical stack trace.

#### Suggested Names:
- `IRContext`
- `ExecutionContext`
- `TraceInfo`

These alternatives might better reflect the data being handled.